lib/fhir_models/bootstrap/hashable.rb

```diff
@@ -69,14 +69,10 @@ def from_hash(hash)
           # handle array of objects
           if value.is_a?(Array)
             value = value.map do |child|
-              obj = child
-              unless [FHIR::RESOURCES, FHIR::TYPES].flatten.include? child.class.name.gsub('FHIR::', '')
-                obj = make_child(child, klass)
-              end
-              obj
+              child.is_a?(FHIR::Model) ? child : make_child(child, klass)
             end
           else # handle single object
-            value = make_child(value, klass)
+            value = make_child(value, klass) unless value.is_a?(FHIR::Model)
```

Comment:
Is there a reason not to make this a guard clause at the beginning of `make_child`? Then this logic wouldn't have to be repeated.